portal-impl/src/com/liferay/portal/language/LanguageResources.java

```diff
@@ -303,7 +303,7 @@ private static Locale _getSuperLocale(Locale locale) {
 			languageMap = Collections.emptyMap();
 		}
 
-		_languageMaps.put(locale, languageMap);
+		_languageMaps.putIfAbsent(locale, languageMap);
```

Comment:
@marianoalvarosaiz How is this change related to the fix?

Tina

Reply:
@marianoalvarosaiz This code is not going to related to race condition. If you take a deep at how this is used, you would see that the language map put in here is always an empty map, and no one is going to modify the map later, all the usages is to read the map. I am going to skip this commit.

Tina.